ssl/build.info

```diff
@@ -32,11 +28,22 @@ SOURCE[../libssl]=\
         ssl_asn1.c ssl_txt.c ssl_init.c ssl_conf.c  ssl_mcnf.c \
         bio_ssl.c ssl_err.c ssl_err_legacy.c tls_srp.c t1_trce.c ssl_utst.c \
         record/ssl3_buffer.c record/ssl3_record.c record/dtls1_bitmap.c \
-        statem/statem.c record/ssl3_record_tls13.c record/tls_pad.c \
+        statem/statem.c record/ssl3_record_tls13.c \
         tls_depr.c $KTLSSRC
 IF[{- !$disabled{'deprecated-3.0'} -}]
-  SOURCE[../libssl]=s3_cbc.c  ssl_rsa_legacy.c
+  SOURCE[../libssl]=ssl_rsa_legacy.c
 ENDIF
+# For shared builds we need to include the libcrypto packet.c and sources
+# needed in providers (s3_cbc.c and record/tls_pad.c) in libssl as well.
+# We can either continue to do it like this, or export all the used
+# symbols so that libssl can use them like any other.
+IF[{- !$disabled{shared} -}]
+  SOURCE[../libssl]=record/tls_pad.c ../crypto/packet.c
+ENDIF
+IF[{- !$disabled{'deprecated-3.0'} && !$disabled{shared} -}]
+  SOURCE[../libssl]=s3_cbc.c
+ENDIF
```

Comment:
Unfortunately, we still produce static libraries *as well* when configured `shared`.  If you only want them to end up in the shared library, use `SHARED_SOURCE` instead of `SOURCE`

Reply:
If you use `SHARED_SOURCE`, the will also *only* be considered for the shared library, and will otherwise be ignored.  This means that you can drop the condition on `$disabled{shared}` too

Reply:
Good point! Thanks for the hint.